addon/mixins/routes/application.js

```diff
@@ -31,6 +31,13 @@ export default Ember.Mixin.create({
         outlet: outlet,
         parentView: parentView
       });
+
+      // Check if another modal exists then give it focus
+      Ember.run.later(function() {
```

Comment:
Ok. Try `Ember.run.next` - that will schedule the focus to take place in the next run loop (after the modal has been removed).